src/pages/index.js

```diff
@@ -36,58 +47,178 @@ const modalImageContainer = document.querySelector(
+    .addingNewCard(formData)
+    .then((cardData) => {
+      const card = createCard(cardData);
+      section.addItem(card);
+      addCardModal.close();
+    })
+    .catch((err) => console.log(err));
+});
+
+const section = new Section(
+  {
+    renderer: (cardData) => {
+      const card = createCard(cardData);
+      section.addItem(card);
+    },
+  },
+  ".cards__list"
+);
+
+api
+  .getInitialCards()
+  .then((data) => {
+    console.log(data);
+
+    section.renderItems(data);
+  })
+  .catch((err) => console.log(err));
+
 /* -------------------------------------------------------------------------- */
-/*                                  Functions                                 */
+/*                                  Functions                                */
 /* -------------------------------------------------------------------------- */
-
-section.renderItems();
+profilePictureModalValidator.enableValidation();
 profileFormValidator.enableValidation();
 addCardFormValidator.enableValidation();
 addCardModal.setEventListeners();
 popupWithImage.setEventListeners();
+profileAvatarModal.setEventListeners();
+profileModal.setEventListeners();
+popupWithImage.setEventListeners();
 
 function handleCardClick(cardData) {
   popupWithImage.open(cardData);
 }
 
+function handleDeleteClick(card) {
+  // Open the confirmation dialog
+
+  popupWithDelete.open(); // Open the delete confirmation modal
+
+  // Define the action to take when the user confirms deletion
+  const deleteAction = () => {
+    return api
+      .deleteCard(card._id) // Use the API to delete the card by its unique _id
+      .then(() => {
+        // Select the card element by its unique _id (ensure it's a valid selector)
+
+        card.removeCard(); // Remove the card from the UI
+        popupWithDelete.close(); // Close the modal after successful deletion
+      })
+      .catch((err) => console.error("Error deleting card:", err)); // Handle any errors during deletion
+  };
+
+  // Set the submit action (deleteAction) for when the modal submit button is clicked
+  popupWithDelete.setSubmitAction(deleteAction);
+}
+
+function handleLikeClick(card) {
+  // Toggle the current like status based on the current state
+  const likeStatus = !card.isLiked; // If the card is liked, make it unliked, and vice versa
+
+  // Send the updated like status to the API
+  api
+    .addRemoveLikes(card._id, likeStatus)
+    .then((data) => {
+      // Update the card's like status locally with the response data
+      card.isLiked = data.isLiked; // Assuming the response contains `isLiked` field or similar
+
+      const cardLikeButton =
+        card._cardElement.querySelector(".card__like-button");
+
+      // Toggle the button's active state based on the new like status
+      if (card.isLiked) {
+        cardLikeButton.classList.add("card__like-button_active");
+      } else {
+        cardLikeButton.classList.remove("card__like-button_active");
```

Comment:
Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle